synapse/lib/layer.py

```diff
@@ -1368,52 +1379,102 @@ def _testAddTagIndx(self, buid, form, tag):
         tagabrv = self.tagabrv.bytsToAbrv(tag.encode())
         self.layrslab.put(tagabrv + formabrv, buid, db=self.bytag)
 
-    async def verify(self, buids=True, tags=True, props=True):
-        if buids:
-            async for error in self.verifyAllBuids():
+    async def verify(self, config=None):
+
+        if config is None:
+            config = {}
+
+        defconf = None
+        if config.get('scanall', True):
+            defconf = {}
+
+        scans = config.get('scans', {})
+
+        nodescan = scans.get('nodes', defconf)
+        if nodescan is not None:
+            async for error in self.verifyAllBuids(nodescan):
                 yield error
 
-        if tags:
-            async for error in self.verifyAllTags():
+        tagsscan = scans.get('tagindex', defconf)
+        if tagsscan is not None:
+            async for error in self.verifyAllTags(tagsscan):
                 yield error
 
-        if props:
-            async for error in self.verifyAllProps():
+        propscan = scans.get('propindex', defconf)
+        if propscan is not None:
+            async for error in self.verifyAllProps(propscan):
                 yield error
 
-    async def verifyAllBuids(self):
+    async def verifyAllBuids(self, scanconf=None):
+        if scanconf is None:
+            scanconf = {}
+
         async for buid, sode in self.getStorNodes():
             async for error in self.verifyByBuid(buid, sode):
                 yield error
 
-    async def verifyAllTags(self):
+    async def verifyAllTags(self, scanconf=None):
+
+        if scanconf is None:
+            scanconf = {}
+
+        globs = None
+
+        includes = scanconf.get('include', ())
+        if includes:
+            globs = s_cache.TagGlobs()
+            for incname in includes:
+                globs.add(incname, True)
+
+        autofix = scanconf.get('autofix')
         for name in self.tagabrv.names():
-            async for error in self.verifyByTag(name):
+
+            if globs is not None and not globs.get(name):
+                continue
+
+            async for error in self.verifyByTag(name, autofix=autofix):
                 yield error
 
-    async def verifyAllProps(self):
+    async def verifyAllProps(self, scanconf=None):
+        if scanconf is None:
+            scanconf = {}
         for form, prop in self.getFormProps():
             async for error in self.verifyByProp(form, prop):
                 yield error
 
-    async def verifyByTag(self, tag):
+    async def verifyByTag(self, tag, autofix=None):
         tagabrv = self.tagabrv.bytsToAbrv(tag.encode())
+
+        async def tryfix(lkey, buid, form):
```

Comment:
I know this is still alpha/beta but how do we want to consider handling mirrors in the event people are fixing things? Make it a manual process?

Reply:
also do we have to declare this inline?

Reply:
All the very autofix stuff is specifically designed to not support mirrors because the mirrors may or may not contain the same errors :)

The inline could be moved into a function that takes several more arguments that are already in scope ( just like every inline function ) but i def dont see any other need for having it in scope.

Reply:
:+1: